### PR TITLE
Storage: Add ability to list files in a bucket.

### PR DESCRIPTION
**Pull request by tjholm**
Add files listing to buckets, a list of files can be retrieved by using:

``` typescript
import { bucket } from '@nitric/sdk';

const myBucket = bucket('my-bucket');

// Get list of files refrerences
const files = await myBucket.files();

// delete all files on the bucket
// this demonstrates that file references are returned.
await Promise.all(files.map(f => f.delete()));
```

### CODE REVIEW

File: src/api/storage/v0/storage.test.ts [hunk 1]
```diff
@@ -341,4 +341,75 @@ describe('Storage Client Tests', () => {
...
+
+          return null as any;
+        });
+    });
+
+    afterAll(() => {
+      jest.resetAllMocks();
+    });
+
+    test('Then StorageClient.delete should reject', async () => {
```
Comment (HomelessDinosaur) [inline src/api/storage/v0/storage.test.ts:367]: should this be StorageClient.ListFiles should reject?

File: src/api/storage/v0/storage.test.ts [hunk 2]
```diff
@@ -341,4 +341,75 @@ describe('Storage Client Tests', () => {
...
+      const client = new Storage();
+      await expect(client.bucket('test').files()).rejects.toEqual(
+        new UnimplementedError("UNIMPLEMENTED")
+      );
+    });
+
+    test('The Grpc client for Storage.delete should have been called exactly once', () => {
```
Comment (HomelessDinosaur) [inline src/api/storage/v0/storage.test.ts:374]: StorageClient.ListFiles

File: src/api/storage/v0/storage.test.ts [hunk 3]
```diff
@@ -341,4 +341,75 @@ describe('Storage Client Tests', () => {
...
+
+    test('Then StorageClient.listFiles should return files', async () => {
+      const client = new Storage();
+      const files = await client.bucket('test').files();
+
+      expect(files).toHaveLength(1);
+      expect(files[0].name).toBe('test/test.txt')
+    });
+
+    test('The Grpc client for Storage.delete should have been called exactly once', () => {
```
Comment (HomelessDinosaur) [inline src/api/storage/v0/storage.test.ts:411]: StorageClient.ListFiles

**Review by HomelessDinosaur — APPROVED** [review]

File: src/api/storage/v0/storage.test.ts (hunk 1 above)
Comment (tjholm) [reply src/api/storage/v0/storage.test.ts]: fixed

File: src/api/storage/v0/storage.test.ts (hunk 3 above)
Comment (tjholm) [reply src/api/storage/v0/storage.test.ts]: fixed.

File: src/api/storage/v0/storage.test.ts (hunk 2 above)
Comment (tjholm) [reply src/api/storage/v0/storage.test.ts]: fixed

**Pull request merged**